File: check_logs.py
import json
import mysql.connector

db = mysql.connector.connect(host='localhost',user='root',password='user',database='bhspl_attendance')
c = db.cursor(dictionary=True)
c.execute("SELECT * FROM raw_logs WHERE emp_id='15241' ORDER BY punch_time DESC LIMIT 10")
print("Sridhar Raw Logs:", json.dumps(c.fetchall(), default=str, indent=2))
c.execute("SELECT * FROM raw_logs ORDER BY punch_time DESC LIMIT 5")
print("Latest 5 Raw Logs:", json.dumps(c.fetchall(), default=str, indent=2))

File: src/main/java/com/bhspl/tools/CheckLogs.java
package com.bhspl.tools;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

public class CheckLogs {
    public static void main(String[] args) {
        String q1 = "SELECT * FROM raw_logs WHERE emp_id='15241' ORDER BY punch_time DESC LIMIT 10";
        String q2 = "SELECT * FROM raw_logs ORDER BY punch_time DESC LIMIT 5";
        try (Connection c = DbHelper.getConnection()) {
            try (PreparedStatement ps = c.prepareStatement(q1); ResultSet rs = ps.executeQuery()) {
                System.out.println("Sridhar Raw Logs:");
                DbHelper.printResultSet(rs);
            }
            try (PreparedStatement ps2 = c.prepareStatement(q2); ResultSet rs2 = ps2.executeQuery()) {
                System.out.println("Latest 5 Raw Logs:");
                DbHelper.printResultSet(rs2);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
